core/tee/tee_svc.c

```diff
@@ -110,111 +110,165 @@ uint32_t tee_svc_sys_nocall(void)
 
 TEE_Result tee_svc_sys_get_property(uint32_t prop, tee_uaddr_t buf, size_t blen)
 {
-	static const char api_vers[] = "1.0";
-	static const char descr[] = "Version N.N";
+	/* API implementation version */
+	char api_vers[] = CFG_TEE_API_VERSION;
+
+	/* Implementation description (implementation-dependent) */
+	char descr[] = "";  /* empty string by default */
+
+	/* Device identifier (unique) */
+	TEE_UUID dev_id;
+
 	/*
-	 * Value 100 means:
-	 * System time based on REE-controlled timers. Can be tampered by the
-	 * REE.  The implementation must still guarantee that the system time
+	 * System time protection level
+	 * 100: System time based on REE-controlled timers (default).
+	 * Can be tampered by the REE
+	 * The implementation must still guarantee that the system time
 	 * is monotonous, i.e., successive calls to TEE_GetSystemTime must
 	 * return increasing values of the system time.
+	 * 1000: System time based on a TEE-controlled secure timer.
+	 * The REE cannot interfere with the system time. It may still
+	 * interfere with the scheduling of TEE tasks, but is not able to
+	 * hide delays from a TA calling TEE_GetSystemTime.
 	 */
-	static const uint32_t sys_time_prot_lvl = 100;
-	static const uint32_t ta_time_prot_lvl = 100;
+	uint32_t sys_time_prot_lvl = 100;
+
+	/* TA persistent time protection level
+	 * 100: Persistent time based on an REE-controlled real-time clock
+	 * and on the TEE Trusted Storage for the storage of origins (default).
+	 * 1000: Persistent time based on a TEE-controlled real-time clock
+	 * and the TEE Trusted Storage.
+	 * The real-time clock MUST be out of reach of software attacks
+	 * from the REE.
+	 */
+	uint32_t ta_time_prot_lvl = 100;
+
+	/* Elliptic Curve Cryptographic support (false by default) */
+	bool crypto_ecc_en;
+
+	/* Trusted storage anti rollback protection level
```

Comment:
Comments are not "kernel style". Should be

    /*
     * Trusted storage anti rollback ...

Same at other places